### DISCUSSION TITLE
2023/01/09/zero-to-chatgpt/

### DISCUSSION THREAD

**Discussion by giscus** [xv44586/giscus#2] [Announcements]
# 2023/01/09/zero-to-chatgpt/

NLP | Machine Learning | Developer

https://xv44586.github.io/2023/01/09/zero-to-chatgpt/

**Comment by alphacute** [comment]
赞！

**Comment by wqw547243068** [comment]
写得很好，补充些细节
- 错别字：低纬度->低维度
- 统计语言模型：基于VSM的语言模型，除了N-Gram，附上典型的LDA、pLSA
- 神经网络语言模型：
  - Bengio 在 2003 年提出的神经概率语言模型(Neural Probabilistic Language Model, NPLM)，克服了n-gram模型里词典开销大、需要做平滑的缺点，并提出用 RNN（循环神经网络）做语言模型
  - VSM升级为词袋模型（CBoW及SG）
  - 2010年，Tomas Mikolov 提出基于 RNN 的语言模型(Recurrent Neural Network Language Model, RNNLM)，RNNLM 根本思想和 NPLM 共通，不同点是词逐个输入到模型，隐藏层有个「记忆功能」单元。
  - 2013年，Tomas Mikolov 再接再厉，利用工程经验，引入层次softmax和负采样，解决了RNNLM训练问题，终于诞生了一个真正意义上的神经网络语言模型 Word2vec，主宰了NLP领域预言模型5年多，直至2018年BERT诞生。

**Reply by xv44586** [reply]
* 错别字已改，感谢指正。
* LDA/pLSA/RNNLM 这些都是“旧时代的残党”，鉴于篇幅，我都省略掉了。

**Comment by alexbase17** [comment]
有个疑问，最终in context learning是如何在模型中生效的呢，比如现在的chatgpt也不需要我们输入shot
